### ISSUE TITLE
LCN Binding 2.X variables wrong values

### ISSUE THREAD

**Issue by nordmanntanne** [openhab/openhab-addons#8151]
I am migration to 2.x binding but i have problems with LCN regulator setpoints and variables:

Number:Temperature M103_Variable1 "Büro Temperatur [%.1f %unit%]" <temperature> {channel="lcn:module:XXXX:YYYY:variable#1"} // Temperature in �C
Number:Temperature M103_R1VarSetpoint "Bodenheizung [%.1f %unit%]" <temperature> {channel="lcn:module:XXXX:YYYY:rvarsetpoint#1"}
Number:Temperature M103_R2VarSetpoint "Wandheizung [%.1f %unit%]" <temperature> {channel="lcn:module:XXXX:YYYY:rvarsetpoint#2"}

I want to read out the current temperature from M103_Variable1 and set temperature with rvarsetpoint.
Sometimes i get wrong values and it is swapping between -100°C and the correct value.
2020-07-18 13:55:36.137 [vent.ItemStateChangedEvent] - M103_Variable1 changed from -100.0 °C to 15.0 °C
2020-07-18 13:55:41.039 [vent.ItemStateChangedEvent] - M103_Variable1 changed from 23.7 °C to -100.0 °C

In LCN PRO Logging don't see this wron (-100°C) values.

**Comment by fwolter** [comment]
-100°C is equivalent to the native LCN value 0. Can you post a snippet from the LCN-PRO bus monitor when the value changes to -100°C?

Does this happen only with Variables or with setpoints too?

Does this happen with all variables?

**Comment by nordmanntanne** [comment]
Hello Fabian,

 

i only use variable 1 or 2 safe temperature from temperature sensor. It is also happening to setpoints.

I attachted both logs. There is nothing in the logs in LCN Pro.

 

Regards,

viel Grüße

Norman


 

Gesendet: Samstag, 18. Juli 2020 um 16:20 Uhr
Von: "Fabian Wolter" <notifications@github.com>
An: "openhab/openhab-addons" <openhab-addons@noreply.github.com>
Cc: "nordmanntanne" <nseeck@gmx.de>, "Author" <author@noreply.github.com>
Betreff: Re: [openhab/openhab-addons] LCN Binding 2.X variables wrong values (#8151)


 

-100°C is equivalent to the native LCN value 0. Can you post a snippet from the LCN-PRO bus monitor when the value changes to -100°C?

Does this happen only with Variables or with setpoints too?

Does this happen with all variables?

**Comment by fwolter** [comment]
Mail attachments don't get through. Can you attach the files in the GitHub web interface?

**Comment by nordmanntanne** [comment]
![LCNPRO](https://user-images.githubusercontent.com/68467135/87879503-825f4280-c9eb-11ea-811a-528bf1ecf1e1.PNG)
![OHLog](https://user-images.githubusercontent.com/68467135/87879508-88552380-c9eb-11ea-9b27-6defb88c6d6f.PNG)

**Comment by fwolter** [comment]
Do you see analogue values at all in the bus monitor? Did you tick "Analogwerte"?
![grafik](https://user-images.githubusercontent.com/1894841/87880078-0cf57100-c9ef-11ea-9077-87daf50156b8.png)

I know this behavior if more than one sensor writes into the same Variable. The sensors overwrite each others values, then. But that should affect all visualizations and regulations in your LCN installation.

**Comment by nordmanntanne** [comment]
Yes i ticked "Analogwerte" but there are no log entries in lcn pro but monitor. I tried copying the variable R1Var to TVar (screenshot) and read out variable1 instead of 2. This solve the issue for now more than 2 days.
![TVar](https://user-images.githubusercontent.com/68467135/88090670-34367480-cb8e-11ea-968c-6aece488ad68.JPG)

**Comment by fwolter** [comment]
I see that your module has the old measurement processing. That's why the values are not shown in the bus monitor. Requesting these values is a bit tricky from the protocol perspective. However, I can't reproduce the behavior you described. Interesting that copying to TVar "solves" the problem. Can you paste all of your item configuration of one module that has these problems? Variable2 seems to be missing in your first post. You mentioned that you use Variable 1 which corresponds to TVar. What is your use case for that?

**Comment by LCN-User** [comment]
Hello,
since it is still open, I have to revisit the topic.
I have switched from OH2 with LCN binding 1.x to OH3 with LCN binding 2.x in the last few days.
Currently, both installations are running on their own RaspberryPi 4B each and are both connected to the LCN bus at the same time via an LCN-PKE.
I also notice that with the new binding I get wrong values from time to time.
For example, I have an item LCN_UHR_VAL that is incremented every 36 seconds by the LCN module itself. It is the well-known LCN imitation of a clock, since my LCN modules are all older.
![Screenshot 2021-01-07-LCN_UHR_VAL](https://user-images.githubusercontent.com/76812252/103912008-0e366980-5107-11eb-8d09-3fe9d8ccef42.jpg)

Here is a snippet from the openHAB Log Viewer with OH2.5 and LCN binding 1.x. Everything looks good...
![Screenshot 2021-01-07_LCN_UHR_VAL_OH2](https://user-images.githubusercontent.com/76812252/103911923-efd06e00-5106-11eb-8a2d-15fb1da33b23.jpg)

Here is a snippet from the openHAB Log Viewer with OH3 and LCN binding 2.x at about the same time. Not so good...
![Screenshot 2021-01-07_LCN_UHR_VAL_OH3](https://user-images.githubusercontent.com/76812252/103909609-36709900-5104-11eb-9e6d-fe3736664392.jpg)

The new binding seems to be not yet as robust as the old one. If the old binding reads the values without errors, the new binding should also do this, regardless of any settings in the modules.

**Comment by fwolter** [comment]
It seems like the Item is updated by two different modules. That is why the value is flapping. The counter of the first module starts at 1615 and the counter of the second module at 1209. This looks like a configuration error. Can you post your complete config?

**Comment by LCN-User** [comment]
There is only one module, module M012 in segment 000, variable 1 (Tvar). 
In both OpenHAB installations there is an item LCN_UHR_VAL that visualizes the value of the same Tvar. The Tvar is only changed in module M012 itself, in OpenHAB it is only displayed.

**Comment by fwolter** [comment]
Can you set the log level to TRACE for the LCN binding and post the output of openhab.log and events.log?

`log:set TRACE org.openhab.binding.lcn`

https://www.openhab.org/docs/administration/logging.html

**Comment by LCN-User** [comment]
The output of the openHAB Log Viewer with the filter "M000012" set:
2021-01-08 10:30:24.875 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01053'
2021-01-08 10:30:34.769 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:30:34.893 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:30:37.912 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:30:38.781 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:30:38.864 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01053'
2021-01-08 10:30:42.920 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:30:43.705 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201208'
2021-01-08 10:30:54.951 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01054'
2021-01-08 10:31:04.883 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:31:05.007 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:31:07.985 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:31:08.895 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:31:08.989 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01054'
2021-01-08 10:31:12.997 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:31:25.024 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01054'
2021-01-08 10:31:34.974 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:31:35.097 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:31:38.986 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:31:39.068 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01055'
2021-01-08 10:31:43.064 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:31:50.317 [TRACE] [g.lcn.internal.connection.Connection] - Received: ':M000012Bx000'
2021-01-08 10:31:55.101 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01055'
2021-01-08 10:31:55.904 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201208'
2021-01-08 10:32:05.071 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:32:08.133 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:32:09.084 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:32:09.166 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01056'
2021-01-08 10:32:13.136 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:32:15.138 [TRACE] [g.lcn.internal.connection.Connection] - Received: ':M000012Rx000'
2021-01-08 10:32:25.168 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01056'
2021-01-08 10:32:35.195 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:32:35.238 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:32:38.207 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:32:39.207 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:32:39.249 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01057'
2021-01-08 10:32:43.207 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:32:55.253 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01057'
2021-01-08 10:33:05.299 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:33:05.432 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:33:08.107 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201208'
2021-01-08 10:33:08.289 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:33:09.312 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:33:09.353 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01057'
2021-01-08 10:33:13.298 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:33:25.336 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01058'
2021-01-08 10:33:35.405 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:33:35.529 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:33:38.359 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:33:39.417 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:33:39.460 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01058'
2021-01-08 10:33:43.372 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:33:55.403 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01059'
2021-01-08 10:34:05.510 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:34:05.664 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:34:08.432 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:34:09.528 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:34:09.580 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01059'
2021-01-08 10:34:13.443 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:34:19.577 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWSA
2021-01-08 10:34:19.667 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:34:20.310 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201208'
2021-01-08 10:34:22.590 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.SMR
2021-01-08 10:34:22.838 [TRACE] [g.lcn.internal.connection.Connection] - Received: ':M000012Rx000'
2021-01-08 10:34:25.468 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01059'
2021-01-08 10:34:35.651 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:34:35.906 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:34:38.518 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:34:39.665 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:34:39.717 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01060'
2021-01-08 10:34:43.535 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:34:43.683 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.SMB
2021-01-08 10:34:43.846 [TRACE] [g.lcn.internal.connection.Connection] - Received: ':M000012Bx000'
2021-01-08 10:34:55.596 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01060' 
2021-01-08 10:35:05.781 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:35:05.974 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:35:08.629 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:35:09.794 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:35:09.856 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01061'
2021-01-08 10:35:13.646 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:35:25.676 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01061'
2021-01-08 10:35:32.511 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201208'
2021-01-08 10:35:35.876 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:35:36.010 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:35:38.706 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:35:39.885 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:35:39.937 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01062'
2021-01-08 10:35:43.711 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:35:55.737 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01062'
2021-01-08 10:36:05.970 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:36:08.769 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:36:09.982 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:36:10.025 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01062'
2021-01-08 10:36:13.794 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:36:25.818 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01063'
2021-01-08 10:36:36.073 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:36:38.854 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:36:40.086 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:36:40.129 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01063'
2021-01-08 10:36:43.866 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:36:44.711 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201208'
2021-01-08 10:36:55.894 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01064'
2021-01-08 10:37:06.181 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-08 10:37:06.295 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:37:08.931 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01208'
2021-01-08 10:37:10.194 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-08 10:37:10.236 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01064'
2021-01-08 10:37:13.951 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01195'
2021-01-08 10:37:25.978 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01064'

**Comment by fwolter** [comment]
Your module uses the old measurement processing with only TVar, R1Var and R2Var. In this case only one application may request measurement values from a module, because the answer of the module cannot be allocated to the previous request. The module answers simply "Measurement value is 1234", but not which variable. The only way OH can allocate the answer to a variable is by knowing which variable it requested before. But if multiple OH instances (or even other modules) request the measurement value, the answer of the module is ambigous.

Try to disable the OH 1.x instance and watch if the error disappears.

**Comment by LCN-User** [comment]
Ok, thanks for the clarification. So if I understood you correctly, here's what happens:
OH3 with LCN binding 2.x asks the module for the value in Tvar and waits for a response. Almost at the same time OH2 with LCN-Binding 1.x asks the same module for the value of another variable, e.g. R1var. Then the module answers only with its module ID and the numerical value without specifying the associated variable name. This then inevitably leads to the errors.
If I see it correctly in the log, you send the PCK command ">M000012.MWV" to request the measured value from the module. In all cases - at least in this log - the first response received is the correct one. Maybe it is possible to make the binding more robust against these errors by using only the first response after a query, at least when using older modules. Possibly then these errors would occur much less often. Maybe there should also be something like a timeout, after sending the query.
How does the 1.x binding do it, since there are no errors in the log, it seems to be more robust against these errors.

**Comment by LCN-User** [comment]
So, as suggested I stopped the OH2 service now with: 
`sudo systemctl stop openhab2`  
After that, only one response appears in the log after each measured value query with ">M000012.MWV" and all is well. This means that when I eventually finish my OH3 installation and finally shut down OH2, the errors should stop at the latest. Still, it would be nice to be able to run a test system in parallel without provoking errors.

Thanks again for your quick response and your patience as well as the understandable explanation. And also a big thank you for the fact that this binding exists at all. Without this binding I could not use openHAB 3 at all.

**Comment by fwolter** [comment]
Great that it works for you now! You understood the issue correctly. Actually the measurement processing was quite the only thing which wasn't touched when migrating the binding from 1.x to 2.x. One could make this part of the code more robust but you will end up with glitches from time to time if you use more than one visualization.

**Comment by fwolter** [comment]
I did some investigations why the OH 3.x binding behaves differently than the 1.x. Indeed there was a bug which causes a mixup of TVar with R1Var/R2Var if multiple softwares/peripherals request the measurement values with modules before 2013.

I also fixed a bug that caused the initial value of R1Var/R2Var setpoints not to be processed with modules before 2013. These were only displayed when changed externally of OH.

If you like to have a JAR with the fixes for older modules, please raise your hand.

**Comment by LCN-User** [comment]
That sounds good, I would like to try that.
My OH3 installation is running on a Raspberry Pi 4B. The installation was done via openhabian 1.6.2. Perhaps you can also briefly explain how the *.jar - file must be integrated into the system. Thank you for your effort and persistence in tracking down the cause of the error. I will observe the thing then times and give a feedback, how it goes.

**Comment by fwolter** [comment]
You can download the JAR here: https://github.com/fwolter/openhab-addons/releases/download/lcn-v0.1.0/org.openhab.binding.lcn-3.1.0-SNAPSHOT.jar

All you need to do is to drop it into the addons folder. I think it is located at `/usr/shar/openhab/addons` in openhabian. OH doesn't need a restart.

**Comment by LCN-User** [comment]
I copied the file to the desired folder yesterday, but I can't see any change in behavior even after a reboot. Is there any way I can check if the new version of the binding is actually running?
I found the previous version of the binding in this folder:
/srv/openhab-userdata/tmp/kar/openhab-addons-3.0.0/org/openhab/addons/bundles/org.openhab.binding.lcn/3.0.0
The file is called there org.openhab.binding.lcn-3.0.0.jar

**Comment by fwolter** [comment]
When you log into the OH [console](https://www.openhab.org/docs/administration/console.html), enter `bundle:list`. You should see something like:

`208 │ Active │  80 │ 3.1.0.202101091827      │ openHAB Add-ons :: Bundles :: LCN Binding`

To be sure the 3.1.0.x version is used, you could uninstall the original LCN binding in the UI.

**Comment by LCN-User** [comment]
Thanks for your quick feedback. Accordingly, I am currently running
250 │ Active │  80 │ 3.1.0.202101101503      │ openHAB Add-ons :: Bundles :: LCN Binding
However, the above incorrect values for the LCN_UHR_VAL item can still be observed.
I have now watched the trace logging for module 12 again for a (longer) time. In all cases, the first response for both the MWV and MWTA requests was the correct one. In one single case (20:59:17.958) a MWTA request remained unanswered. The other responses in the log are probably attributable to the other OH instance and should be ignored by the binding.
Here is again a short excerpt from the log:
2021-01-11 20:53:54.825 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:53:54.868 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02082'
2021-01-11 20:54:05.024 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:54:06.026 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:54:16.896 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:54:16.938 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:54:22.150 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02082'
2021-01-11 20:54:24.926 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:54:24.980 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02083'
2021-01-11 20:54:35.094 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:54:36.097 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:54:47.001 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:54:47.043 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:54:48.510 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201211'
2021-01-11 20:54:52.387 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02083'
2021-01-11 20:54:55.027 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:54:55.070 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02083'
2021-01-11 20:55:05.330 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:55:06.340 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:55:17.100 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:55:17.142 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:55:22.456 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02084'
2021-01-11 20:55:25.129 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:55:25.182 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02084'
2021-01-11 20:55:35.408 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:55:40.426 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:55:47.205 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:55:47.257 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:55:52.533 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02085'
2021-01-11 20:55:55.235 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:55:55.287 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02085'
2021-01-11 20:56:00.704 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201211'
2021-01-11 20:56:05.479 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:56:10.492 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:56:17.309 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:56:17.351 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:56:22.565 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02086'
2021-01-11 20:56:25.337 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:56:25.379 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02086'
2021-01-11 20:56:35.565 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:56:40.570 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:56:47.410 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:56:47.463 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:56:52.636 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02087'
2021-01-11 20:56:55.441 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:56:55.483 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02087'
2021-01-11 20:57:05.632 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:57:10.643 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:57:12.909 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201211'
2021-01-11 20:57:17.511 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:57:17.555 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:57:25.536 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:57:25.588 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02088'
2021-01-11 20:57:35.714 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:57:40.719 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:57:47.618 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:57:47.670 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:57:52.785 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02088'
2021-01-11 20:57:55.651 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:57:55.703 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02088'
2021-01-11 20:58:05.791 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:58:10.805 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:58:17.762 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:58:17.815 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:58:22.863 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02089'
2021-01-11 20:58:25.113 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201211'
2021-01-11 20:58:25.789 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:58:25.841 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02089'
2021-01-11 20:58:35.886 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:58:40.908 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:58:47.857 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:58:47.900 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:58:52.972 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02090'
2021-01-11 20:58:55.890 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:58:55.943 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02090'
2021-01-11 20:59:05.974 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:59:10.987 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'
2021-01-11 20:59:17.958 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:59:21.979 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWTA
2021-01-11 20:59:22.083 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:59:23.054 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02091'
2021-01-11 20:59:25.992 [TRACE] [g.lcn.internal.connection.Connection] - Sent: >M000012.MWV
2021-01-11 20:59:26.044 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.02091'
2021-01-11 20:59:36.051 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01211'
2021-01-11 20:59:37.312 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.A00201211'
2021-01-11 20:59:41.058 [TRACE] [g.lcn.internal.connection.Connection] - Received: '%M000012.01205'

**Comment by fwolter** [comment]
You should see the following message if a measurement value has been received, which wasn't requested.
`15:40:31.084 [-thingHandler-3] WARN  o.o.b.l.i.s.LcnModuleVariableSubHandler:77 - S000M022: Received a measurement value which has not been requested. Reading measurement values by more than one software/peripheral is not supported by modules before 2013.`

Maybe I somehow didn't manage to compile the right version for you. Here is a fresh version https://github.com/fwolter/openhab-addons/releases/download/lcn2021-01-13/org.openhab.binding.lcn-3.1.0-SNAPSHOT.jar

**Comment by LCN-User** [comment]
I have now copied the new file again to the /usr/share/openhab/addons folder and restarted the RaspberryPi, but I still cannot see any change in behavior.  I can't find any warning messages like above in the TRACE log either.
I then looked in the OH console again and ran the bundle:list command.
In doing so, I noticed that the LCN binding is in the list twice.
215 │ Active │ 80 │ 3.0.0 │ openHAB Add-ons :: Bundles :: LCN Binding
250 │ Active │ 80 │ 3.1.0.202101131445 │ openHAB Add-ons :: Bundles :: LCN Binding
Is this correct?

**Comment by fwolter** [comment]
You can uninstall the LCN binding via the UI to be sure that the new version is used. The most recent version of the JAR can be found here https://github.com/fwolter/openhab-addons/releases/download/lcn2021-19-01/org.openhab.binding.lcn-3.1.0-SNAPSHOT.jar It contains also a fix from another issue concerning old measurement value processing.

**Comment by LCN-User** [comment]
Sorry it took a few days, but today I had some time to play around with openHAB. 
I downloaded the new .jar file and copied it to the above mentioned directory /usr/share/openhab/addons. And indeed, it's the first time that I can't observe any erroneous values for my LCN_UHR_VAL item.
Thank you very much for your great support and unwavering patience.

**Comment by fwolter** [comment]
Great that it is working now!
